scipy/io/tests/test_wavfile.py

```diff
@@ -84,9 +84,18 @@ def test_read_fail():
         fp.close()
 
 
-def test_read_early_eof():
+def test_read_early_eof_with_data():
     for mmap in [False, True]:
         fp = open(datafile('test-44100Hz-le-1ch-4bytes-early-eof.wav'), 'rb')
+        rate, data = wavfile.read(fp, mmap=mmap)
+        fp.close()
+
+        del data
+
+
+def test_read_early_eof():
+    for mmap in [False, True]:
+        fp = open(datafile('test-44100Hz-le-1ch-4bytes-early-eof-no-data.wav'), 'rb')
```

Comment:
I suppose we may wish to consider using a context manager here, though I do see that the surrounding code still uses an explicit `close()` call instead--perhaps the original code is pretty old.

Reply:
@tylerjereddy The `open` and `close` statements from the other tests looks to be 8-10 years (e.g. 61c61e9f783d8813d36354f6e4f1584dc7a7b01f).  I can change it on the test I've made and the one I've modified, but perhaps the others should have their own issue/PR?

To reiterate a side note, opening a file and passing that in instead of passing the filename to `wavread` bypasses the `mmap`.